x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -512,6 +523,13 @@ export class TaskRunner<
       schedule: taskSchedule,
     } = this.taskInstance;
 
+    if (apm.currentTransaction) {
+      apm.currentTransaction.name = `Execute Alerting Rule`;
```

Comment:
Did you consider setting the name of the rule in the transaction name? Something like `"Execute Alerting Rule: $alerting_rule_name"` ?

Reply:
Understood, it's done on https://github.com/elastic/kibana/pull/117504/files#diff-27ff694dc534e9e509d43b00c563eabada722db2371c9bcf016ff3c8b929c5fbR488